sql/core/src/test/scala/org/apache/spark/sql/streaming/StreamingQueryListenerSuite.scala

```diff
@@ -133,6 +133,93 @@ class StreamingQueryListenerSuite extends StreamTest with BeforeAndAfter {
     }
   }
 
+  testQuietly("multiple listeners, check trigger events are generated correctly") {
```

Comment:
I'm ok with that (with a small fix)
need to add:
listeners.foreach(listener => spark.streams.addListener(listener))